### ISSUE TITLE
Issue using latest azure storage (>= 7.1.2 and even >= 7.0.0)

### ISSUE THREAD

**Issue by onionhammer** [dotnet/orleans#1912]
Orleans seems incompatible with WindowsAzure.Storage 7.1.2 

```
at Orleans.Runtime.Host.AzureSilo.Start(ClusterConfiguration config, String deploymentId, String connectionString)
Exc level 1: Orleans.Runtime.OrleansException: Exception trying to create or connect to the Azure table: The remote server returned an error: (415) Unsupported Media Type.
at Orleans.AzureUtils.OrleansSiloInstanceManager.<GetManager>d__14.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
c:\Program Files (x86)\Jenkins\workspace\release_dotnet_master\Lib\ClassLibraryCommon\Table\CloudTable.cs:line 1183
```

**Comment by jdom** [comment]
Without having investigated, I wonder if it's related to this:
> Changes in 7.1.2 :
- All: Support for 2015-12-11 REST version. Please see our REST API documentation and blogs for information about the related added features. If you are using the Storage Emulator, please update to Emulator version 4.4.

From their [changelog](https://github.com/Azure/azure-storage-net/blob/master/changelog.txt)

**Comment by jdom** [comment]
Ping @centur who also ran into the same issue

**Comment by jdom** [comment]
Relevant information, from [Versioning for the Azure Storage Services](https://msdn.microsoft.com/en-us/library/azure/dd894041.aspx)
> Beginning with version 2015-12-11, the Atom feed is no longer supported as a payload format for Table service operations. Version 2015-12-11 and later versions support only JSON for the payload format.

It is related to they dropping support for that media type. We need to figure out how to address this

**Comment by jkonecki** [comment]
I can confirm the issue. Workaround is to downgrade Storage nuget package to 7.0.0

**Comment by benjaminpetit** [comment]
I submitted #1968 to support WindowsAzure.Storage 7.0+

Azure made a breaking change in the version 7.0 of their library by [removing an enum value from TablePayloadFormat](https://github.com/Azure/azure-storage-net/commit/b4699d38f6583a2cb4136586e687f04c7784fae8#diff-f2dcdff30895201b510b4347e27c46ed). As they don’t enforce enum order, they basically shifted all of them.

If you use version 7.0, the [Accept header is not set](https://github.com/Azure/azure-storage-net/blob/master/Lib/WindowsRuntime/Table/Protocol/TableOperationHttpRequestFactory.cs#L240), and by default the **service will think you want atomPub**.

On version 7.1.2, they switched to the 2015-12-11 REST api version, that doesn’t accept atomPub anymore, so it fails.

**Comment by jdom** [comment]
BTW, @jkonecki, from @benjaminpetit's explanation, and being a little bit more explicit, it looks like downgrading to 7.0.0 *works* although it does unintentionally use AtomPub for metadata. If you use AzureStorage < 7.0 instead then Orleans either explicitly disables metadata or uses JSON metadata which is more compact. Not sure about how important this is, it's just that basically it still works by luck, but with slightly different behavior than what we expected to use, and there could in theory be other behavioral changes that we do not know of.
Hence why we decided for Orleans to depend on version 7.0.0 and avoid other potential issues. According to semantic versioning, you will be able to build your app against 7.1.2 using binding redirects, and it will now work as expected (we don't want to force people using 7.0.0 to upgrade to the latest for no apparent reason).

**Comment by jdom** [comment]
BTW, this is the specific place that changed the behavior if you use binding redirects from a version previous to 7.0.0: https://github.com/Azure/azure-storage-net/pull/252/files#diff-f2dcdff30895201b510b4347e27c46edL31  from PR Azure/azure-storage-net#252.

Ideally the removal of that enum value should have not shifted all the other enum values by being explicit about the values they take, but it wasn't the case.

**Comment by sergeybykov** [comment]
Resolved via #1968.
